go/merkletreev2/config.go

```diff
@@ -0,0 +1,53 @@
+package merkletreev2
+
+import "fmt"
+
+// Config defines the shape of the MerkleTree.
+type Config struct {
+	// A hasher is used to compute hashes in this configuration
+	hasher Hasher
+
+	// The number of children per node. Must be a power of two.
+	childrenPerNode uint
+
+	// The maximum number of KeyValuePairs in a leaf node before we split
+	valuesPerLeaf uint
+
+	// The number of bits necessary to represent a ChildIndex, i.e. log2(m)
+	bitsPerIndex uint8
```

Comment:
Is this derived from the other values? If so, probably want at least a test to make sure it's always in alignment.